src/operator/roi_pooling.cc

```diff
@@ -227,15 +227,56 @@ Operator *ROIPoolingProp::CreateOperatorEx(Context ctx, std::vector<TShape> *in_
 DMLC_REGISTER_PARAMETER(ROIPoolingParam);
 
 MXNET_REGISTER_OP_PROPERTY(ROIPooling, ROIPoolingProp)
-.describe("Performs region-of-interest pooling on inputs. Resize bounding box coordinates by "
-"spatial_scale and crop input feature maps accordingly. The cropped feature maps are pooled "
-"by max pooling to a fixed size output indicated by pooled_size. batch_size will change to "
-"the number of region bounding boxes after ROIPooling")
-.add_argument("data", "NDArray-or-Symbol", "Input data to the pooling operator, a 4D Feature maps")
+.describe(R"code(Performs region of interest(ROI) pooling on the input array.
+
+ROI pooling is a variant of max pooling layer, in which output size is fixed and
+input rectangle is a parameter.  Its purpose is to perform max pooling on the inputs
+of non-uniform sizes to obtain fixed-size feature maps. ROI pooling is a neural-net
+layer mostly used in training a `Fast R-CNN` network for object detection tasks.
+
+This operator takes a 4D feature map as an input array and region proposals as rois,
+then it pools over sub-regions of input and produces a fixed sized output array
+regardless of the ROI size.
+
+To crop input feature map accordingly, you can resize the bounding box coordinates
+by changing the parameters `rois` and `spatial_scale`.
+
+The cropped feature maps are pooled by standard max pooling operation to a fixed size output
+indicated by a `pooled_size` parameter. batch_size will change to the number of region
+bounding boxes after ROIPooling.
+
+The size of the region of interest(`rois`) doesn’t have to be perfectly divisible by
```

Comment:
there is a mismatch. you say "the region of interest" but in parenthesis you have rois (regions of interest). Is it singular or plural? Do you mean "The number of regions of interest? Or the size of each region?